completions/ip

```diff
@@ -87,19 +100,12 @@ _comp_cmd_ip()
                     # TODO
                     ;;
                 delete)
-                    case $((cword - subcword)) in
-                        1)
-                            _comp_compgen_available_interfaces
-                            ;;
-                        2)
-                            _comp_compgen -- -W 'type'
-                            ;;
-                        3)
-                            [[ $prev == type ]] &&
-                                _comp_compgen -- -W 'vlan veth vcan dummy ifb
-                                    macvlan can'
-                            ;;
-                    esac
+                    _comp_compgen_available_interfaces
+                    if [[ $prev == type ]]; then
+                        _comp_cmd_ip__link_types "$1"
```

Comment:
Could you explain those changes in behavior?

- Why was the check for the word position removed? Hmm, I tried to find a description in `man ip` but cannot find the description on the arguments to `delete`.
- The completions generated by `_comp_compgen_available_interfaces` seem to be overwritten by `_comp_cmd_ip__link_types`, but then why `_comp_compgen_available_interfaces` is called outside the `if` statement? How is it different from calling `_comp_compgen_available_interfaces` inside the `false` branch of the `if` statement?

Reply:
Ah, OK. For the first point, I noticed the explanation in the commit message. Thanks.

Reply:
You're correct, I didn't notice it. I moved `_comp_compgen_available_interfaces` to the else branch, since it only needs to be called than.

Reply:
Thanks for updating!